zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/ZooKeeperEndpointGroup.java

```diff
@@ -67,58 +70,80 @@ public ZooKeeperEndpointGroup(String zkConnectionStr, String zNodePath, int sess
      */
     public ZooKeeperEndpointGroup(String zkConnectionStr, String zNodePath, int sessionTimeout,
                                   NodeValueCodec nodeValueCodec) {
-        zooKeeperConnector = new ZooKeeperConnector(zkConnectionStr, zNodePath, sessionTimeout,
-                                                    createListener());
+        checkArgument(!isNullOrEmpty(zNodePath), "zNodePath");
+        checkArgument(sessionTimeout > 0, "sessionTimeout: %s (expected: > 0)",
+                      sessionTimeout);
         this.nodeValueCodec = requireNonNull(nodeValueCodec, "nodeValueCodec");
-        zooKeeperConnector.connect();
-    }
-
-    @Override
-    public void close() {
-        zooKeeperConnector.close(true);
+        this.internalClient = true;
+        this.client = CuratorFrameworkFactory.builder()
+                                             .connectString(zkConnectionStr)
+                                             .retryPolicy(Constants.DEFAULT_RETRY_POLICY)
+                                             .sessionTimeoutMs(sessionTimeout)
+                                             .build();
+        client.start();
+        pathChildrenCache = new PathChildrenCache(client, zNodePath, true);
+        pathChildrenCache.getListenable().addListener((c, event) -> {
+            switch (event.getType()) {
+                case CHILD_ADDED:
+                    addEndpoint(this.nodeValueCodec.decode(event.getData().getData()));
+                    break;
+                case CHILD_REMOVED:
+                    removeEndpoint(this.nodeValueCodec.decode(event.getData().getData()));
+                    break;
+                default:
+                    break;
+            }
+        });
+        try {
+            pathChildrenCache.start();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
     }
 
     /**
-     * Create a {@link ZooKeeperListener} listens specific ZooKeeper events.
-     * @return  {@link ZooKeeperListener}
+     * Create a ZooKeeper-based {@link EndpointGroup}, endpoints will be retrieved from a node's all children's
+     * node value using {@link NodeValueCodec}.
+     * @param client the curator framework instance.
```

Comment:
- Please add more spaces between `client` and `the curator` so that it's aligned with other parameters
- curator -> Curator
- Please insert an empty line before the first `@param`.
- Please remove the trailing dot.